src/WattleScript.Interpreter/Execution/VM/FunctionProto.cs

```diff
@@ -4,18 +4,19 @@
 namespace WattleScript.Interpreter.Execution.VM
 {
     [Flags]
-    enum FunctionFlags
+    internal enum FunctionFlags
     {
         None = 0x0,
         IsChunk = 0x1,
         TakesSelf = 0x2,
         ImplicitThis = 0x4
     }
-    class FunctionProto
+    
+    public class FunctionProto
     {
         //Function Data
         public string Name;
```

Comment:
All these exposed fields should be changed to a readonly public property + an internal field.  (the arrays that need to be exposed having `IReadOnlyList<T>` accessors), as the C# side should not be allowed to modify function prototypes.

Reply:
Fixed in c8591eac35b0ac1afcc87910dd92832cea2c6262. Note that I've had to rename internal fields to lowercase. Added XML docs for the public properties